flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/delegation/PlannerBase.scala

```diff
@@ -176,11 +176,11 @@ abstract class PlannerBase(
         val input = getRelBuilder.queryOperation(modifyOperation.getChild).build()
         val identifier = catalogManager.qualifyIdentifier(catalogSink.getTablePath: _*)
         getTableSink(identifier).map(sink => {
-          TableSinkUtils.validateSink(catalogSink, identifier, sink)
+          val partKeys =
+            catalogManager.getTable(identifier).get().asInstanceOf[CatalogTable].getPartitionKeys
+          TableSinkUtils.validateSink(catalogSink, identifier, sink, partKeys)
           sink match {
-            case partitionableSink: PartitionableTableSink
-              if partitionableSink.getPartitionFieldNames != null
-                && partitionableSink.getPartitionFieldNames.nonEmpty =>
+            case partitionableSink: PartitionableTableSink =>
```

Comment:
see comment on `BatchExecSink`, should we just put the validation here? (For partitioned table, we need to make sure the sink is `PartitionableTableSink`)

Reply:
Yes, we can.

Reply:
I think we have validateSink above, no need to validate here.